pycue/opencue/wrappers/host.py

```diff
@@ -217,6 +220,177 @@ def setThreadMode(self, mode):
+        # Workstations are marked as Nimby, ignore those.
+        hosts = [host for host in hosts if not host.isNimbyEnabled() and host.isUp()]
+        check_hosts_interval_seconds = 30
+
+        if len(hosts) == 0:
+            print("No hosts found")
+            return
+
+        print("Rebooting hosts:\n%s.\n\n"
+              "Are you sure hosts on this list are safe to be rebooted? [Y/n]" %
+              [str(host.name()) for host in hosts])
+        while True:
+            choice = input()
+            if choice == "Y":
+                break
+            if choice == "n":
+                return
+
+        groups = [hosts[x:x+group_size] for x in range(0, len(hosts), group_size)]
+
+        host_still_rebooting = []
+
+        if start_time is None:
+            start_time = int(time.time())
+        for group in groups:
+            group_host_names = []
+            for host in group:
+                if host.bootTime() < start_time:
+                    try:
+                        host.rebootWhenIdle()
+                    except grpc.RpcError as rpc_error:
+                        # pylint: disable=no-member
+                        if rpc_error.code() in [grpc.StatusCode.UNAVAILABLE,
+                                                grpc.StatusCode.CANCELLED]:
+                            # Wait and Retry
+                            time.sleep(5)
+                            host.rebootWhenIdle()
+                        else:
+                            raise rpc_error
+                        # pylint: enable=no-member
+
+                group_host_names.append(str(host.name()))
+            print("Requesting reboot for %s. Waiting for completion." % group_host_names)
+
+            # Wait until 80% of a group gets upgraded to jump to the next group
+            while True:
+                try:
+                    hosts = search.HostSearch.byName(group_host_names)
+                    rebooting_hosts = [str(host.name()) for host in hosts
+                                       if Host.hasHostRebootedSince(host, start_time)]
+                    if len(rebooting_hosts) <= len(hosts) * 0.2:
+                        host_still_rebooting.extend(rebooting_hosts)
+                        print("Moving to the next group, left behind %s..\n" % rebooting_hosts)
+                        break
+                    print("Still waiting on %s.." % rebooting_hosts)
+                    time.sleep(check_hosts_interval_seconds)
+                except grpc.RpcError as rpc_error:
+                    # Ignore rpc unavailable to survive short service outages on the server side
+                    # pylint: disable=no-member
+                    if rpc_error.code() in [grpc.StatusCode.UNAVAILABLE, grpc.StatusCode.CANCELLED]:
+                        continue
+                    # pylint: enable=no-member
+
+        # Wait for remaining hosts to finish (20% not awaited for)
+        if len(host_still_rebooting) > 0:
```

Comment:
Suggestion:
`if host_still_rebooting:`: This is a more Pythonic way to check if `host_still_rebooting` is not empty. In Python, an empty list evaluates to False, and a non-empty list evaluates to True.

Reply:
Yeah, I pretty much dislike this in python. But I agree it's better to conform with the language. 
```suggestion
        if host_still_rebooting:
```